tests/test_features/test_projection.py

```diff
@@ -0,0 +1,209 @@
+from ..fixtures import Dataset
+from unittest import mock
+
+from sklearn.datasets import make_classification, make_regression
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import  StandardScaler
+from sklearn.decomposition import PCA
+from sklearn.manifold import LocallyLinearEmbedding
+
+##########################################################################
+## Fixtures
+##########################################################################
+
+@pytest.fixture(scope="class")
+def discrete(request):
+    """
+    Creare a random classification fixture.
+    """
+    X, y = make_classification(
+        n_samples=400, n_features=12, n_informative=10, n_redundant=0,
+        n_classes=5, random_state=2019)
+
+    # Set a class attribute for discrete data
+    request.cls.discrete = Dataset(X, y)
+
+@pytest.fixture(scope="class")
+def continuous(request):
+    """
+    Creates a random regressor fixture.
+    """
+    X, y = make_regression(
+        n_samples=500, n_features=22, n_informative=8, random_state=2019
+    )
+
+    # Set a class attribute for continuous data
+    request.cls.continuous = Dataset(X, y)
+
+@pytest.fixture(scope="function")
+def transformer(request):
+    """
+    Creates a PCA transformer. Acccepts the projection dimnesion as param and
+    returns transformer accordingly. If parame is not specified then None is returned.
+    """
+    if (not hasattr(request, "param")):
+        return
+    pca_transformer = Pipeline([("scale", StandardScaler()),
+                                    ("pca", PCA(request.param, random_state=2019))])
+    # Set a class attribute for transformer
+    request.cls.transformer = pca_transformer
+
+
+@pytest.mark.usefixtures("discrete", "continuous", "transformer")
+class TestProjectionVisualizerBase(VisualTestCase):
+    
+    @pytest.mark.parametrize("transformer", [2], indirect=True)
+    def test_discrete_plot(self):
+
+        X, y = self.discrete
+        classes = ["a", "b", "c", "d", "e"]
+        visualizer = ProjectionVisualizer(projection=2,
+                                          colormap="plasma", classes=classes)
+        visualizer.transformer=self.transformer
+        transform_array = visualizer.fit_transform(X, y)
+        assert(visualizer.classes_ == classes)
+        visualizer.finalize()
+        self.assert_images_similar(visualizer)
+        assert transform_array.shape == (self.discrete.X.shape[0], 2)
+        
+    def test_continuous_plot(self):
+
+        X, y = self.continuous
+        manifold = Pipeline([
+            ("pca", PCA(n_components=15, random_state=1998)),
+            ("lle", LocallyLinearEmbedding(n_components=2, random_state=2019)),
+        ])
+
+        visualizer = ProjectionVisualizer(projection="2d")
+        visualizer.transformer = manifold
+        visualizer.fit_transform(X, y)
+        visualizer.finalize()
+        visualizer.cax.set_yticklabels([])
+        self.assert_images_similar(visualizer)
```

Comment:
Thats great! But I think that changing random state can fix it. Or maybe changing the transformer. Because it is working in one of test for continuous.

Reply:
What do you think can be the reason of difference?

Reply:
I think it's just the windows renderer vs the gtk renderer - it doesn't seem to be a random state issue, but I'll keep looking for a place where there is randomness that we didn't fix.